### ISSUE TITLE
Add Leap Wallet

### ISSUE THREAD

**Issue by mhagel** [hicommonwealth/commonwealth#5378]
## Description

- Integrate Leap wallet

## Engineering Requirements

- Keep in mind One Signer, One Account tickets. This can be done before or after #5078 , but we need to remember that with multiple wallets supported, it is possible to login with Keplr or Leap using the same address or "signer".

## Additional context

https://docs.cosmoskit.com/
https://docs.leapwallet.io/cosmos/for-dapps-connect-to-leap/add-leap-to-existing-keplr-integration

- Follow-up: Add Leap's metamask snap to the Cosmoskit wallet list: #5379 
- deferring CosmosKit - nice to have but would require a fair amount of refactoring.
